Individual locales files

ferleal opened the pull request:
Added Option to create Multiple Files using --multi flag so you can have language files dependent of its view 
Notice I use a independent file for errors and words to prevent duplicates on my language but still have a single component js to load , I change the Locale getting the $noExt  like this 

```php
App::setLocale($noExt); 
```
en/login.php and es/login.php

```php
return [

    /*
    |--------------------------------------------------------------------------
    | Login Language Lines
    |--------------------------------------------------------------------------
    | Login form
    |
    */

    'email' => [
        'label' => Lang::get('words.email'),
        'error' => Lang::get('errors.please_enter_valid',['label' => Lang::get('words.email')]),
    ],
 ];
```

login.js
```js
"en": {
        "email": {
            "label": "Email Address",
            "error": "Please enter a valid Email Address",
        },
      :
    },
    "es": {
        "email": {
            "label": "Correo Electronico",
            "error": "Por favor inserta un Correo Electronico correcto",
        },
    } 
```

and the final js can be imported via component 
login.vue
```vue
<template><label for="email">{{ $t('email.label') }}</label></template>
<script type="text/javascript">
  import loginLang from '../../langs/login';
export default {
data() {
      return {
      };
    },
    i18n: { // `i18n` option

      messages: loginLang ,
    },
</script>
}
```

ferleal committed:
Added Option to create Multiple Files with multiple languages so it can be implemented via Component based localization

ferleal committed:
Added option to change Locales so we use Lang:get() on our language files on laravel and recive a pure js fully translated , example in php         'error' => Lang::get('errors.please_enter_valid',['label' => Lang::get('words.email')]),  on js
"en": {
        "email": {
            "error": "Please enter a valid Email Address",
        },
    },
    "es": {
        "email": {
            "error": "Por favor inserta un Correo Electronico correcto",
        },
    }

martinlindhe commented on src/Generator.php:
```diff
@@ -39,22 +43,81 @@ public function generateFromPath($path, $umd = null)
+     * @return string
+     * @throws Exception
+     */
+    public function generateMultiple($path, $umd = null)
```
i think this should be private

ferleal replied:
oh but if we make it private it will not be accessible from GenerateInclude

martinlindhe replied:
Ah:-) good then!

martinlindhe commented on src/Generator.php:
```diff
@@ -98,35 +162,39 @@ private function allocateLocaleArray($path)
                     continue;
                 }
+
+                $root = realpath(base_path() . config('vue-i18n-generator.langPath') . '/' . $lastLocale);
```
since generateMultiple() is conditionally called earlier, and generateMultiple() initializes availableLocales, the  $lastLocale will be empty here if generateMultiple() wasn't called.

I think that is what is causing travis to fail at the moment.

martinlindhe commented on src/Generator.php:
```diff
@@ -98,35 +162,39 @@ private function allocateLocaleArray($path)
      * @return array
      */
-    private function adjustArray(array $arr)
+    private
```
not sure about multiline function declarations

ferleal replied:
oops it was phpstorm reformat code, fixed

martinlindhe commented:
It seems the change broke the default case. You can trigger the tests by running "vendor/phpunit/phpunit/phpunit"

See my review comments above )

ferleal committed:
Fixed multi line on privates
Added validation on allocateLocaleArray to prevent  creation of fileToCreate if no Locale is defined

martinlindhe commented:
Looks great, CI is happy now! Thanks for your contribution!
